src/EditorFeatures/CSharp/BlockCommentEditing/BlockCommentEditingCommandHandler.cs

```diff
@@ -0,0 +1,177 @@
+    [Order(After = PredefinedCommandHandlerNames.Completion)]
+    internal class BlockCommentEditingCommandHandler : AbstractBlockCommentEditingCommandHandler
+    {
+        [ImportingConstructor]
+        public BlockCommentEditingCommandHandler(
+            ITextUndoHistoryRegistry undoHistoryRegistry,
+            IEditorOperationsFactoryService editorOperationsFactoryService) : base(undoHistoryRegistry, editorOperationsFactoryService)
+        {
+        }
+
+        protected override string GetExteriorTextForNextLine(SnapshotPoint caretPosition)
+        {
+            var currentLine = caretPosition.Snapshot.GetLineFromPosition(caretPosition);
+
+            var firstNonWhitespacePosition = currentLine.GetFirstNonWhitespacePosition() ?? -1;
+            if (firstNonWhitespacePosition == -1)
+            {
+                return null;
+            }
+
+            var currentLineStartsWithBlockCommentStartString = currentLine.StartsWith(firstNonWhitespacePosition, "/*", ignoreCase: false);
+            var currentLineStartsWithBlockCommentEndString = currentLine.StartsWith(firstNonWhitespacePosition, "*/", ignoreCase: false);
+            var currentLineStartsWithBlockCommentMiddleString = currentLine.StartsWith(firstNonWhitespacePosition, "*", ignoreCase: false);
+
+            if (!currentLineStartsWithBlockCommentStartString && !currentLineStartsWithBlockCommentMiddleString)
+            {
+                return null;
+            }
+
+            if (!IsCaretInsideBlockCommentSyntax(caretPosition))
+            {
+                return null;
+            }
+
+            if (currentLineStartsWithBlockCommentStartString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*|*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      /|*
+                    return null; // The newline inserted could break the syntax in a way that this handler cannot fix, let's leave it.
+                }
+                else
+                {
+                    //      /*|
+                    return " *" + GetPaddingOrIndentation(currentLine, caretPosition, firstNonWhitespacePosition, "/*");
+                }
+            }
+
+            if (currentLineStartsWithBlockCommentEndString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*
+                    //      |*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      *|/
+                    return "*";
+                }
+                else
+                {
+                    //      /*
+                    //   |   */
+                    return " * ";
+                }
```

Comment:
One space padding for the caret position. It's simple logic and always return one space thus save some string concat. So that `GetPaddingOrIndentation` can simply assume the caret position was behind the `*` and save an extra `if` branching here.